homeassistant/components/honeywell/__init__.py

```diff
@@ -1 +1,117 @@
 """Support for Honeywell (US) Total Connect Comfort climate systems."""
+from datetime import timedelta
+
+import somecomfort
+
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.util import Throttle
+
+from .const import _LOGGER, CONF_DEV_ID, CONF_LOC_ID, DOMAIN
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=180)
+PLATFORMS = ["climate"]
+
+
+async def async_setup_entry(hass, config):
+    """Set up the Honeywell thermostat."""
+    username = config.data[CONF_USERNAME]
+    password = config.data[CONF_PASSWORD]
+
+    client = await hass.async_add_executor_job(
+        get_somecomfort_client, username, password
+    )
+
+    if client is None:
+        return False
+
+    loc_id = config.data.get(CONF_LOC_ID)
+    dev_id = config.data.get(CONF_DEV_ID)
+
+    for location in client.locations_by_id.values():
+        for device in location.devices_by_id.values():
+            if (not loc_id or location.locationid == loc_id) and (
+                not dev_id or device.deviceid == dev_id
+            ):
+                data = HoneywellService(hass, client, username, password, device)
+                await data.update()
+                hass.data[DOMAIN] = data
+                hass.config_entries.async_setup_platforms(config, PLATFORMS)
+
+    return True
+
+
+def get_somecomfort_client(username, password):
+    """Initialize the somecomfort client."""
+    try:
+        return somecomfort.SomeComfort(username, password)
+    except somecomfort.AuthError:
+        _LOGGER.error("Failed to login to honeywell account %s", username)
+        return None
```

Comment:
raise ConfigEntryAuthFailed once reauth is implemented (in a followup PR)

Reply:
Will do.